cody.el

```diff
@@ -797,48 +886,38 @@ there is a connection."
     (setq cody--node-version-status nil) ; re-check node version on start
     (message "Initializing Cody connection...")
     (cody--connection)
-    (message "Cody connection initialized.")
-    (when cody-auto-enable-cody-mode
-      (cody--init-cody-mode))))
-
-(defun cody--init-cody-mode ()
-  "Start an agent if needed, and enable Cody in applicable buffers."
-  ;; TODO: Add a customization option to not auto-enable cody-mode
-  ;; for all buffers, and instead the user can invoke cody-mode manually
-  ;; or set up their own rules for toggling it. As one example driving this
-  ;; use case, the user might only want to have a chat session open, but
-  ;; not want Cody tracking all their other buffers.
-  (if cody--use-threads
-      ;; N.B. Debugger is unable to exit if there's an error in the thread function.
-      (make-thread (lambda ()
-                     (cody--enable-cody-mode))
-                   "Cody startup thread")
-    (cody--enable-cody-mode)))
+    (message "Cody connection initialized.")))
 
 (defun cody-restart ()
   "Shut down and restart Cody.  Mostly for debugging."
   (interactive)
-  (let ((cody--node-version-status 'good))
-    (ignore-errors
-      (cody-shutdown)))
-  (setq cody--node-version-status nil)
-  (cody-start))
+  (let ((buffers (cl-loop for buf in (buffer-list)
+                          when (with-current-buffer buf cody-mode)
+                          collect buf)))
+    (let ((cody--node-version-status 'good))
```

Comment:
Worth a helper function? Short term, will explain why; long term, will make our lives easier when there's a second thing we want to ignore while we "just do it." It would be nice if we ended up with a canonical set of those things instead of people doing them ad-hoc.